src/bindgen/ir/generic_path.rs

```diff
@@ -265,7 +311,11 @@ impl GenericPath {
         self.ctype = resolver.type_for(&self.path);
     }
 
-    pub fn load(path: &syn::Path) -> Result<Self, String> {
+    pub fn set_assoc(&mut self, assoc: Option<AssocTypeId>){
+        self.assoc = assoc;
+    }
+
+    pub fn load(path: &syn::Path, qself: &Option<syn::QSelf>) -> Result<Self, String> {
```

Comment:
Changed because syn::ExprPath parsing (the data in an enum variant of syn::Expr::Path(ExprPath) also calls GenericPath::load. However, it uses syn::ExprPath instead of syn::TypePath.